src/components/elements/table/index.js

```diff
@@ -1,14 +1,67 @@
+import { Component } from 'react'
 import factory from '../factory'
-import cx from 'classnames'
 import Header from './header'
 import Body from './body'
 
-export default ({ api, property }) => (
-  <div className='ct-table-container'>
-    <label className='ct-element-label ct-table-label'>{property.title}</label>
-    <table className='ct-table'>
-      <Header api={api} property={property} />
-      <Body api={api} property={property} />
-    </table>
-  </div>
-)
+export default class Table extends Component {
+  constructor(props) {
+    super()
+
+    this.state = {
+      sortedRows: props.property.rows.slice(),
+      currentColumnIndex: null,
+    }
+  }
+
+  render() {
+    return (
+      <div className='ct-table-container'>
+        <label className='ct-element-label ct-table-label'>{this.props.property.title}</label>
+        <table className='ct-table'>
+          <Header
+            api={this.props.api}
+            columns={this.props.property.columns}
+            sortBy={index => this.sortBy(index)} />
+          <Body rows={this.state.sortedRows} />
+        </table>
+      </div>)
+  }
+
+  sortBy(columnIndex) {
+    const rowsToSort = this.state.sortedRows
+    let sorted = rowsToSort.sort((row1, row2) => {
+      if (this.getRowValue(row1, columnIndex) > this.getRowValue(row2, columnIndex)) return 1
+      if (this.getRowValue(row1, columnIndex) < this.getRowValue(row2, columnIndex)) return -1
+      return 0
+    })
+    if (this.state.currentColumnIndex == columnIndex) {
+      sorted = sorted.reverse()
+    }
+    this.setState({
+      sortedRows: sorted,
+      currentColumnIndex: this.state.currentColumnIndex == columnIndex ? null : columnIndex,
+    })
+    return false
+  }
+
+  getRowValue(row, columnIndex) {
+    if (row[columnIndex].type == 'html') {
+      const doc = document.implementation.createHTMLDocument('')
+      doc.documentElement.innerHTML = row[columnIndex].content
+
+      return doc.getElementsByTagName('body')[0].innerText
```

Comment:
This is pretty smart! If performance is an issue, one alternative could be to just strip out the tags with a regex (`<[^>]*>`, the way @npehrsson would have liked it!).